lib/thor/line_editor/readline.rb

```diff
@@ -1,12 +1,12 @@
+begin
+  require "readline"
+rescue LoadError
+end
+
 class Thor
   module LineEditor
     class Readline < Basic
       def self.available?
-        begin
-          require "readline"
```

Comment:
@deivid-rodriguez wasn't this added because of bundler? if we move out to the top of the file would not that mean that bundler will require readline too early?

Reply:
This change isn't necessary to get CI green (it was only so I could run one spec independently). If it's blocking the merge I can just remove this commit from the PR.

Reply:
Yes, I added this because of Bundler. I don't recall the specifics, but it was something related to Windows I believe. I'd prefer if this commit was removed, yes.

I did reproduce the failing spec in isolation, but that can be fixed in the spec itself without giving up on requiring deadline lazily:

```diff
diff --git a/spec/line_editor_spec.rb b/spec/line_editor_spec.rb
index f034ec8..9c6250e 100644
--- a/spec/line_editor_spec.rb
+++ b/spec/line_editor_spec.rb
@@ -1,8 +1,9 @@
 require "helper"
+require "readline"
 
 describe Thor::LineEditor, "on a system with Readline support" do
   before do
-    @original_readline = ::Readline if defined? ::Readline
+    @original_readline = ::Readline
     silence_warnings { ::Readline = double("Readline") }
   end
 
@@ -22,10 +23,8 @@ end
 
 describe Thor::LineEditor, "on a system without Readline support" do
   before do
-    if defined? ::Readline
-      @original_readline = ::Readline
-      Object.send(:remove_const, :Readline)
-    end
+    @original_readline = ::Readline
+    Object.send(:remove_const, :Readline)
   end
 
   after do
```

Reply:
Is the failure what's fixed by #771?

Reply:
Ah, ok! In that case I think it makes sense to merge this, and I can work on the fix for the ruby-head issue separately.

Reply:
Also fixed running line editor specs in isolation at https://github.com/rails/thor/pull/791!